synapseclient/client.py

```diff
@@ -2589,7 +2585,7 @@ def _download_from_URL(
                     else:
                         mode = "wb"
                         previouslyTransferred = 0
-                        sig = hashlib.md5()
+                        sig = hashlib.new("md5", usedforsecurity=False)
```

Comment:
@linglp can you speak to the error this is trying to fix?

Reply:
Yeah.. If I don't make the change, I would get an error like this: 
```
[main]  INFO    running on Python 3.11.3
Run started:2023-06-27 22:10:20.620752

Test results:
>> Issue: [B324:hashlib] Use of weak MD5 hash for security. Consider usedforsecurity=False
   Severity: High   Confidence: High
   CWE: CWE-327 (https://cwe.mitre.org/data/definitions/327.html)
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b324_hashlib.html
   Location: synapseclient/client.py:2588:30
2587                            previouslyTransferred = 0
2588                            sig = hashlib.md5()
2589

--------------------------------------------------
>> Issue: [B324:hashlib] Use of weak MD5 hash for security. Consider usedforsecurity=False
   Severity: High   Confidence: High
   CWE: CWE-327 (https://cwe.mitre.org/data/definitions/327.html)
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b324_hashlib.html
   Location: synapseclient/core/utils.py:53:10
52
53          md5 = hashlib.md5()
54          with open(filename, "rb") as f:

--------------------------------------------------

```